code/obj/misc_junk.dm

```diff
@@ -852,3 +852,86 @@
 		next_artifact.set_loc(place.loc)
 		processing_items.Remove(src)
 		..()
+
+/obj/item/scpgnome
+	name = "strange sarcophagus"
+	desc = "A sarcophagus bound by magical chains."
+	icon = 'icons/obj/junk.dmi'
+	icon_state = "sarc_0"
+	density = 1
+	var/gnome = 1
+
+	attackby(obj/item/W as obj, mob/user as mob)
+		if(istype(W,/obj/item/scpgnome_lid) && ((src.icon_state == "sarc_2")||(src.icon_state == "sarc_3")))
+			user.u_equip(W)
+			qdel(W)
+			src.icon_state = "sarc_1"
+		else if(istype(W,/obj/item/gnomechompski/mummified) && (src.icon_state == "sarc_3"))
+			user.u_equip(W)
+			qdel(W)
+			src.icon_state = "sarc_2"
+			src.gnome = 1
+		else if(istype(W,/obj/item/device/key/chompskey) && (src.icon_state == "sarc_0"))
+			user.u_equip(W)
+			qdel(W)
+			src.icon_state = "sarc_key"
+		else
+			..()
+
+	attack_hand(mob/user as mob)
+		if(src.icon_state == "sarc_key")
+			src.icon_state = "opening"
+			sleep(23)
+			src.icon_state = "sarc_1"
```

Comment:
When it comes to changing icon states with a delay I got a cool trick!
```suggestion
			animate(src, time = 2.3 SECONDS)
			animate(icon_state = "sarc_1")
```
This basically adds a 2.3 second long animation which does nothing and queues up an instant animation that changes the icon state right after that. The benefit is that this is all done clientside and there are no `sleep`s or `spawn`s.